Fixes #25329: Fixed return table in module docs generated by plugin_formatter.py.

andy-maier opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
This change fixes two issues with the return table generated by `plugin_formatter.py`:
1. When specifying a list of strings in the 'description' field of a return value, it shows them in Python list syntax on the resulting web page, e.g. `['a', 'b', 'c']`.
2. When specifying more than one line for the 'sample' field, the result table gets damaged in the HTML output.

In addition, this change re-arranges the HTML tags produced in the generated RST file such that they line up nicely and can better be checked by humans for completeness.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
- `docs/templates/plugin.rst.j2`

##### ANSIBLE VERSION
- 2.4 devel

andy-maier committed:
Fixed return table in module docs generated by plugin_formatter.py (#25329)

This change fixes two issues with the generated return table:

1. When specifying a list of strings in the 'description' field of a
   return value, it shows them in Python list syntax on the resulting
   web page, e.g. `['a', 'b', 'c']`.
2. When specifying more than one line for the 'sample' field, the
   result table gets damaged in the HTML output.

In addition, this change re-arranges the HTML tags produced in the
generated RST file such that they line up nicely and can better be
checked by humans for completeness.

Signed-off-by: Andreas Maier <andreas.r.maier@gmx.de>

sivel commented on docs/templates/plugin.rst.j2:
```diff
@@ -148,52 +178,65 @@ Common return values are documented here :doc:`common_return_values`, the follow
+    <td>
+{% for desc in returndocs[entry].description %}
+        <div>@{ desc | replace('\n', '\n    ') | html_ify }@</div>
+{% endfor %}
```
The schema defines this field as either a string or a list of strings.  Wouldn't this break the instance where it is just a string?  We would be iterating each individual character of that string.

andy-maier replied:
You are correct. I just verified that it does. Sorry for this. I'll fix that.

andy-maier replied:
Fixed in commit 6842bae.
That commit adds support for description being both a string and a list of strings. It does that for the descriptions of the input parameters and for the description of return values.

andy-maier commented:
Updated the PR description, using the template.

andy-maier commented:
I have made one more small fix (there was an `i` appended to the name of module input parameters that have suboptions, thanks to `vi` ;-)
I squashed my three commits on this PR and rebased to the latest `devel` branch.

andy-maier commented:
Rebased.

dharmabumstead reviewed (approved):
Looks good to me. Someone from the core dev team, please review.

calfonso commented:
rebuild_merge
